File: src/main/java/org/sopt/like/domain/Like.java
package org.sopt.like.domain;

import org.sopt.common.domain.BaseTimeEntity;
import org.sopt.post.domain.Post;
import org.sopt.user.domain.User;

import jakarta.persistence.*;

@Entity
@Table(name = "likes")
public class Like extends BaseTimeEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "user_id", nullable = false)
	private User user;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "post_id", nullable = false)
	private Post post;

	protected Like() {}

	public Like( User user, Post post) {
		this.user = user;
		this.post = post;
	}
}

File: src/main/kotlin/org/sopt/like/domain/Like.kt
package org.sopt.like.domain

import jakarta.persistence.*
import org.sopt.common.domain.BaseTimeEntity
import org.sopt.post.domain.Post
import org.sopt.user.domain.User

@Entity
@Table(name = "likes")
class Like : BaseTimeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Long? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private var user: User? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id", nullable = false)
    private var post: Post? = null

    protected constructor()

    constructor(user: User, post: Post) {
        this.user = user
        this.post = post
    }
}
